### ISSUE TITLE
(Ios) Error Module 'Mapbox' not found 

### ISSUE THREAD

**Issue by chaipipatP** [nitaliano/react-native-mapbox-gl#1402]
**Install With CocoaPods**

**Podfile**
```
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'xxxx' do
  rn_path = '../node_modules/react-native'
  rn_maps_path = '../node_modules/react-native-maps'
  pod 'RNSound', :path => '../node_modules/react-native-sound'

  pod 'yoga', path: "#{rn_path}/ReactCommon/yoga/yoga.podspec"
  pod 'React', path: rn_path, subspecs: [
    'Core',
    'RCTActionSheet',
    'RCTAnimation',
    'RCTGeolocation',
    'RCTImage',
    'RCTLinkingIOS',
    'RCTNetwork',
    'RCTSettings',
    'RCTText',
    'RCTVibration',
    'RCTWebSocket',
    'BatchedBridge'
  ]

  pod 'react-native-maps', path: rn_maps_path

  # pod 'GoogleMaps'  # Remove this line if you don't want to support Google Maps on iOS
pod 'react-native-mapbox-gl', :path => '../node_modules/@mapbox/react-native-mapbox-gl'

end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    if target.name == 'react-native-google-maps'
      target.build_configurations.each do |config|
        config.build_settings['CLANG_ENABLE_MODULES'] = 'No'
      end
    end
    if target.name == "React"
      target.remove_from_project
    end
  end
end
```

**Error**
```
/@mapbox/react-native-mapbox-gl/ios/RCTMGL/CameraStop.h:9:9: fatal error: module 'Mapbox' not found
@import Mapbox;
 ~~~~~~~^~~~~~
1 error generated.


** BUILD FAILED **

```
react-native-cli: 2.0.1
react-native: 0.51.0

**Comment by giantss** [comment]
@chaipipatP same issue here，Have you solved the problem?

**Comment by giantss** [comment]
I solved this problem.
Follow the steps below：

1. `yarn add @mapbox/react-native-mapbox-gl`

2. `react-native link`
If this step is finished, running `react-native run-ios` directly will report undel this error.found` this error. The official documentation does not give the corresponding detailed steps.

3. `react-native-mapbox-gl/example/node_modules/@mapbox/react-native-mapbox-gl/ios/Mapbox.framework` add `Link Binary with Libraries`

Looks like this:
![121540372433_ pic_hd](https://user-images.githubusercontent.com/13778798/47425445-e1633980-d7bc-11e8-92a8-92e9cace2df6.jpg)

Then run `react-native run-ios`  will not report an error again. But running the program will crash.
`Reason: image not found`

4. 
<img width="829" alt="image" src="https://user-images.githubusercontent.com/13778798/48605714-820dd900-e9b8-11e8-8756-21992d95c384.png">
<img width="836" alt="image" src="https://user-images.githubusercontent.com/13778798/48605807-c7320b00-e9b8-11e8-98fd-da28e92ab658.png">

finally, Looks like this:
<img width="765" alt="image" src="https://user-images.githubusercontent.com/13778798/48605834-d913ae00-e9b8-11e8-9e9c-a414624a0972.png">



Then run `react-native run-ios`  will not report an error again.

**Comment by FibreFoX** [comment]
@giantss Shouldn't be `pod install` be enough to have this included?

Anyway, I'm facing a strange issue. On a fresh installed machine, this does not happen using XCode 10, on another machine with the SAME code this produces the mentioned error above :( is there anything to solve this issue? (like some hidden lock-file which I need to remove, as the native mapbox SDK seems to be downloaded somewhere ...)

**Comment by giantss** [comment]
@FibreFoX I started to think the same as you, but it was not as perfect as I thought.
I have not upgraded to `xcode 10` , I am using `xcode 9.4` .
 I have not encountered such a problem. Sorry.

**Comment by FibreFoX** [comment]
@giantss no worries :) just wanted to state that the provided solution via "configuration via clicking" is against the idea of using that `Podfile` (at least IMHO).

If it helps, we are using  RN 0.57.4 and `@mapbox/react-native-mapbox-gl` in version `6.1.3`.

**Comment by giantss** [comment]
@FibreFoX I upgraded my environment today and encountered the same problem as you. #1424

**Comment by alburdette619** [comment]
I'm seeing this issue just today on App Center.  Going to try @giantss linking procedures though I thought I had it all done correctly.

**Comment by estevaolucas** [comment]
I'm getting the same error, but just on CircleCi. Locally (from Xcode or Fastlane) it works fine, but when it builds with CircleCi+Fastlane, on both Xcode 9.3 and 10.1.0, id doesn't work out. 

Manual installation or via podfile, the same error happens:
```
▸ /Users/distiller/project/packages/my-project/node_modules/@mapbox/react-native-mapbox-gl/ios/RCTMGL/RCTMGLCallout.h:11:9: fatal error: module 'Mapbox' not found
▸ @import Mapbox;
```

Any idea of what can be done on Fastlane or CircleCi to make it works?

-------
**Update:** 

After almost 20 different trial and error, and all related to module/framework resolving on Xcode - following basically StackOverflow posts [like this](https://stackoverflow.com/questions/29500227/getting-error-no-such-module-using-xcode-but-the-framework-is-there), I realized something that was not clear: when I look into [ios folder](https://github.com/mapbox/react-native-mapbox-gl/tree/master/ios) I don't see `Mapxbox.framework`. Then I saw that [`Mapxbox.framework` is installed from an `npm` `preinstall` hook](https://github.com/mapbox/react-native-mapbox-gl/blob/master/package.json#L29). So taking a look in the logs I figured out that this was not being called somehow from CircleCi. 

(I don't know yet why this npm hook works fine locally, but doesn't work for CircleCi - maybe it's a Lerna issue).

A not ideal solution that I found for it: call mapbox's script to download the framework manually before the ios build (`npm explore @mapbox/react-native-mapbox-gl -- npm run fetch:ios:sdk`).

**Comment by onitzschke** [comment]
Are you using pods?

**Comment by kristfal** [comment]
We got this tracked in a duplicate issue and a PR open to use pods. Expect this to become easier very soon.

**Comment by ashfurrow** [comment]
@kristfal that's great to hear – can you link to the duplicate issue?

**Comment by ashfurrow** [comment]
I was experiencing this issue as well, and was surprised CocoaPods wasn't handling this. Turns out I was on CocoaPods 1.5.0; upgrading to 1.6.1 solved the issue for me.

**Comment by davidxmoody** [comment]
I just spent a couple of hours struggling with pretty much the same issue using `@react-native-mapbox-gl/maps` (and RN 0.61). Here are some brief notes, in case it helps anyone else in a similar situation. It's mostly the same as what estevaolucas said above but with a little bit of extra context.

- In CircleCI, we split our workflow into separate jobs for installing npm deps then building on iOS, building on Android, deploying to TestFlight, etc.
- The npm install step was running on a Linux machine (and the iOS build step was running on a Mac)
- The `npm run fetch:ios:sdk` preinstall hook calls another script `scripts/download-mapbox-gl-native-ios-if-on-mac.js`
- That script checks the platform and will only download the Mapbox SDK if it's running on a Mac
- Because our npm install step was running on Linux, the Mapbox SDK never got downloaded
- To fix it, I ran `npm explore @react-native-mapbox-gl/maps -- npm run fetch:ios:sdk` in the job running on a Mac
